tls/s2n_connection.h

```diff
@@ -305,6 +305,14 @@ struct s2n_connection {
 
     /* Key update data */
     unsigned key_update_pending:1;
+
+    /* Bitmap to represent preferred list of keyshare for client to generate and send keyshares in the ClientHello message.
+     * The least significant bit (lsb), if set, indicates that the client must send an empty keyshare list.
+     * Each bit value in the bitmap indiciates the corresponding curve in the ecc_preferences list for which a key share needs to be generated.
+     * The order of the curves represented in the bitmap is obtained from the security_policy->ecc_preferences.
+     * Setting and manipulating this value requires security_policy to be configured prior.
+     * */
+    uint8_t preferred_key_shares;
```

Comment:
The LSB==empty trick is clever, but for readability we should really just have a separate flag. The LSB behavior is not intuitive.

Reply:
However, since we are just using this API for testing and will revisit it before we make it final, I'm fine leaving this behavior for now.

Reply:
Agreed, we will be re-visiting this API prior to moving it from testing mode.